README.md

```diff
@@ -332,71 +332,98 @@ Jsonnet has the concept of hidden fields. These are fields, that are not going t
 These are the available fields with their respective default values:
 ```
 {
-	_config+:: {
-    namespace: "default",
-
-    versions+:: {
-        alertmanager: "v0.17.0",
-        nodeExporter: "v0.18.1",
-        kubeStateMetrics: "v1.5.0",
-        kubeRbacProxy: "v0.4.1",
-        prometheusOperator: "v0.30.0",
-        prometheus: "v2.10.0",
-    },
-
-    imageRepos+:: {
-        prometheus: "quay.io/prometheus/prometheus",
-        alertmanager: "quay.io/prometheus/alertmanager",
-        kubeStateMetrics: "quay.io/coreos/kube-state-metrics",
-        kubeRbacProxy: "quay.io/brancz/kube-rbac-proxy",
-        nodeExporter: "quay.io/prometheus/node-exporter",
-        prometheusOperator: "quay.io/prometheus-operator/prometheus-operator",
-    },
-
-    prometheus+:: {
-        names: 'k8s',
-        replicas: 2,
-        rules: {},
+  values:: {
+    common: {
+      namespace: 'default',
+      ruleLabels: {
+        role: 'alert-rules',
+        prometheus: $.values.prometheus.name,
+      },
+      // to allow automatic upgrades of components, we store versions in autogenerated `versions.json` file and import it here
+      versions: {
+        alertmanager: error 'must provide version',
+        blackboxExporter: error 'must provide version',
+        grafana: error 'must provide version',
+        kubeStateMetrics: '1.9.8',
+        nodeExporter: error 'must provide version',
+        prometheus: error 'must provide version',
+        prometheusAdapter: error 'must provide version',
+        prometheusOperator: error 'must provide version',
+      } + (import 'versions.json'),
```

Comment:
Maybe we should add another sub-section about version numbers we are using and link to versions.json file there? That file is meant to be automatically updated and now it holds all component versions.

Reply:
Alright I made a second attempt now. :slightly_smiling_face: Removed all the defaults and just linked to `main.libsonnet` and `versions.json` instead. Let me know what you think.